File: problem9.c
// Project Euler problem 9: https://projecteuler.net/problem=9
// Author: Tom Harkness
// May 29, 2021

#include <stdbool.h>
#include <stdio.h>

#define TARGET_SUM 1000

int main(void) {
    for (int a = 1; a < TARGET_SUM; a++) {
        for (int b = 1; b < TARGET_SUM; b++) {
            for (int c = 1; c < TARGET_SUM; c++) {
                if (a*a + b*b - c*c == 0) {
                    if (a + b + c == 1000) {
                        printf("%d", a*b*c);
                        return 0;
                    }
                }
            }   
        }
    }
}

File: python/problem9.py
# Project Euler problem 9: https://projecteuler.net/problem=9
# Author: Tom Harkness
# May 29, 2021

if __name__ == "__main__":
    for a in range(1,800):
        for b in range(a,800):
            for c in range(b,800):
                if a + b + c == 1000:
                    if a**2 + b**2 == c**2:
                        print(a*b*c)
                        exit()